ArmaForces.Boderator/ArmaForces.Boderator.BotService/Controllers/DLCController.cs

```diff
@@ -0,0 +1,162 @@
+                Response.WriteAsync("User not found");
+                return;
+            }
+
+            foreach (var user in usersOnServer)
+            {
+                var objUser = new JObject();
+                objUser.Add("UserID", user.UserID);
+
+                var dlcArray = new JArray();
+
+                foreach (var dlc in user.DLCList)
+                {
+                    var objDLC = new JObject();
+                    objDLC.Add("DLC", dlc.ToString());
+                    dlcArray.Add(objDLC);
+                }
+                objUser.Add("DLC List", dlcArray);
+
+                usersArray.Add(objUser);
+            }
+
+            Response.WriteAsync($"{usersArray.ToString()}");
+        }
+
+        //Finds users wiht given DLC
+        [HttpGet("{dlcName}/users")]
+        public void UsersWithDLC(String dlcName)
+        {
+            var usersArray = new JArray();
+            var usersOnServer = _users.UsersList;
+
+            SingleUser.DLC dlcClass;
+
+            try 
+            {
+                dlcClass = (SingleUser.DLC)Enum.Parse(typeof(SingleUser.DLC), dlcName);
+            }
+            catch
+            {
+                Response.StatusCode = 404;
+                Response.WriteAsync("DLC not found");
+                return;
+            }
+
+            foreach (var user in usersOnServer)
+            {
+                if (user.DLCList.Contains(dlcClass))
+                {
+                    var objUser = new JObject();
+                    objUser.Add("UserID", user.UserID);
+                    usersArray.Add(objUser);
+                }
+            }
+
+            Response.WriteAsync($"{usersArray.ToString()}");
+        }
+
+        //Check if user has all DLCs
+        [HttpGet("DLC")]
+        public IActionResult UserhasDLC(ulong userID, String dlcs)
+        {
+            if (dlcs == null)
+            {
+                return (IActionResult)NotFound("No given DLC");
+            }
+
+            List<SingleUser.DLC> dlcList = new List<SingleUser.DLC>();
+            
+            foreach (string singleDlc in dlcs.Split(','))
+            {
+
+                try
+                {
+                    var dlcClass = (SingleUser.DLC)Enum.Parse(typeof(SingleUser.DLC), singleDlc);
+                    dlcList.Add(dlcClass);
+                }
+                catch
+                {
+                    return (IActionResult)BadRequest("Wrong DLC name");
+                }
+
+            }
+
+            var user = _users.UsersList.SingleOrDefault(x => x.UserID == userID);
+            
+            if (user is null)
+            {
+                return  (IActionResult)NotFound("User not found");
+            }
+
+            var dlcFound = !dlcList.Except(user.DLCList).Any();
+
+            return dlcFound
+                ? Ok("User has all given DLCs")
+                : (IActionResult)NotFound("User doesn't have all given DLCs");
+        }
```

Comment:
This is probably useless in API. Instead just make sure API can return user and his DLCs.